pika/spec.py

```diff
@@ -126,14 +126,7 @@ def synchronous(self):
 
         def decode(self, encoded, offset=0):
             (self.client_properties, offset) = data.decode_table(encoded, offset)
-            length = struct.unpack_from('B', encoded, offset)[0]
-            offset += 1
-            self.mechanism = encoded[offset:offset + length]
-            try:
-                self.mechanism = str(self.mechanism)
-            except UnicodeEncodeError:
-                pass
-            offset += length
+            self.mechanism, offset = data.decode_short_string(encoded, offset)
```

Comment:
Was the old code machine-generated? If it was, then future re-generation will result in loss of mods.